### ISSUE TITLE
Use X-PINNED:true for pinned tasks

### ISSUE THREAD

**Issue by googol42** [dmfs/opentasks#942]
Hello,

in Nextcloud you can pin tasks like you can pin tasks in opentasks. Nextcloud sets `X-PINNED:true` for pinned tasks. I know that this is not part of the spec. However, opentasks seesm to store this information only locally and does not sync it.

I do not know if there are other custom `X-...` attributes in other app for pinning, but I'd like to suggest do support `X-PINNED` in opentasks to sync the pinned status.

In short, the **request** is to use `X-PINNED:true` for pinned tasks.

Cheers

**Comment by dmfs** [comment]
I think you're mistaken here. OpenTasks doesn't sync anything. It only provides the task database and a UI to manage tasks. Synchronization is provided by another app (like our own CalDAV-Sync app).
So the sync app is responsible for providing a mapping between the OpenTasks database and the synchronization data format (iCalendar, in this case). After introducing pinned Tasks in OpenTasks I chose to support it in CalDAV-Sync by adding the custom `X-PINNED` property. To my knowledge this property didn't exist before. NextCloud apparently adopted the property.
Please ask the developer of your sync app to get support for this.

**Comment by googol42** [comment]
But when the `X-PINNED` is not let through by the sync provider, how can it be that I can edit a task (with `X-PINNED` set by nextcloud) with opentasks and the X-PINNED is still there afterwards? This reminds me of #839 where you said the same to me (but the behaviour was different). Anyways, I reach out to davx.

**Comment by dmfs** [comment]
I don't know about DavX's internal workings, so I can't comment on that. I know how OpenTasks and our CalDAV-Sync app work. For me, the pinned status propagates well to my server and my other client devices.
